L1Trigger/VertexFinder/src/VertexFinder.cc

```diff
@@ -438,14 +443,157 @@ namespace l1tVertexFinder {
   }
 
   void VertexFinder::findPrimaryVertex() {
-    double vertexPt = 0;
-    pv_index_ = 0;
+    if (settings_->vx_precision() == Precision::Emulation) {
+      pv_index_ = std::distance(verticesEmulation_.begin(),
+                                std::max_element(verticesEmulation_.begin(),
+                                                 verticesEmulation_.end(),
+                                                 [](const l1t::VertexWord& vertex0, const l1t::VertexWord& vertex1) {
+                                                   return (vertex0.pt() < vertex1.pt());
+                                                 }));
+    } else {
+      pv_index_ = std::distance(
+          vertices_.begin(),
+          std::max_element(
+              vertices_.begin(), vertices_.end(), [](const RecoVertex<>& vertex0, const RecoVertex<>& vertex1) {
+                return (vertex0.pt() < vertex1.pt());
+              }));
+    }
+  }
+
+  // Possible Formatting Codes: https://misc.flogisoft.com/bash/tip_colors_and_formatting
+  template <class data_type, typename stream_type>
+  void VertexFinder::printHistogram(stream_type& stream,
+                                    std::vector<data_type> data,
+                                    int width,
+                                    int minimum,
+                                    int maximum,
+                                    std::string title,
+                                    std::string color) {
+    int tableSize = data.size();
+
+    if (maximum <= minimum) {
+      maximum = float(*std::max_element(std::begin(data), std::end(data))) * 1.05;
+      minimum = float(*std::min_element(std::begin(data), std::end(data)));
+    }
+
+    if (minimum < 0) {
+      minimum *= 1.05;
+    } else {
+      minimum = 0;
+    }
+
+    if (maximum == -1) {
+      maximum = float(*std::max_element(std::begin(data), std::end(data))) * 1.05;
+    }
```

Comment:
Thank you Cecile for taking comments into account.
However, your update won't deal correctly with the case when `maximum = -1 ` (and minimun is above from the parameter set.
A possible suggestion could be:
```suggestion
    if (maximum == -1) {
      maximum = float(*std::max_element(std::begin(data), std::end(data))) * 1.05;
    }
    else if (maximum <= minimum) {
      maximum = float(*std::max_element(std::begin(data), std::end(data))) * 1.05;
      minimum = float(*std::min_element(std::begin(data), std::end(data)));
    }

    if (minimum < 0) {
      minimum *= 1.05;
    } else {
      minimum = 0;
    }
```